src/Compute/Compute/VirtualMachine/Config/AddAzureVMNetworkInterfaceConfigurationCommand.cs

```diff
@@ -0,0 +1,140 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Compute.Common;
+using Microsoft.Azure.Commands.Compute.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.Azure.Management.Internal.Network.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    /// <summary>
+    /// Setup the network interface.
+    /// </summary>
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VMNetworkInterfaceConfiguration"), OutputType(typeof(PSVirtualMachine))]
+    public class AddAzureVMNetworkInterfaceConfigurationCommand : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
```

Comment:
This is a good addition.

Although we should create a new proposal for this  so that it can be approved.

I had initially left this out to reduce the work, and since the powershell cmdlet already did implicit creation all on its own.

But now that it is written good to have.

Reply:
Maybe we should split these undesign reviewed changes into a different PR, so we are not blocked for check in on the design review?